pydantic/dataclasses.py

```diff
@@ -69,8 +72,11 @@ def _process_class(
     config: Type['BaseConfig'],
 ) -> 'DataclassType':
     post_init_original = getattr(_cls, '__post_init__', None)
+    post_init_post_parse = getattr(_cls, '__post_init_post_parse__', None)
     if post_init_original and post_init_original.__name__ == '_pydantic_post_init':
         post_init_original = None
+    if post_init_post_parse:
```

Comment:
this is wrong, it needs to be removed completely, I guess this is why tests are failing.